### PR TITLE
DT-66: Remove old headers that aren't needed and upgrade

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,82 +1,108 @@
...
+  environment:
+    JAVA_TOOL_OPTIONS: -Xmx1024m -XX:ParallelGCThreads=2 -XX:ConcGCThreads=2 -XX:ParallelGCThreads=2 -Djava.util.concurrent.ForkJoinPool.common.parallelism=2
 
-      # Specify service dependencies here if necessary
-      # CircleCI maintains a library of pre-built images
-      # documented at https://circleci.com/docs/2.0/circleci-images/
-      # - image: circleci/postgres:9.4
-
-    working_directory: ~/repo
+version: 2.1
```
Comment (petergphillips) [inline .circleci/config.yml]: switch to circle 2.1 :grinning:

Comment (petergphillips) [reply .circleci/config.yml]: @mattops what do you think?

File: Dockerfile [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
 
+RUN addgroup --gid 2000 --system appgroup && \
+    adduser --uid 2000 --system appuser --gid 2000
+
+WORKDIR /app
+
+COPY --chown=appuser:appgroup build/libs/custody-api*.jar /app/app.jar
+COPY --chown=appuser:appgroup run.sh /app
+
+USER 2000
```
Comment (timharrison-moj) [inline Dockerfile]: Could use the newly created username rather than uid? Just thinking it might be clearer, but a small thing.

File: build.gradle [hunk 3]
```diff
@@ -50,66 +50,53 @@ springBoot {
...
-    compile 'org.apache.commons:commons-lang3:3.8.1'
-    compile 'net.logstash.logback:logstash-logback-encoder:5.2'
-    compile 'com.microsoft.azure:applicationinsights-spring-boot-starter:1.2.0-BETA'
-    compile 'com.microsoft.azure:applicationinsights-logging-logback:2.3.1'
-
-    annotationProcessor 'org.projectlombok:lombok:1.18.4'
+    implementation files('lib/ojdbc8-12.2.0.1.jar')
+
+    implementation 'com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.9.9'
+    implementation 'com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.9.9'
```
Comment (timharrison-moj) [inline build.gradle]: I saw the jdk8 version of the jackson-datatype package and checked it - it is deprecated, as described here 
 https://github.com/FasterXML/jackson-datatype-jdk8 - but might be a bigger job to unpick it?  Happy if we leave this for another PR - I know the point of this one is to make a start on standardising with other services.

File: Dockerfile (hunk 2 above)
Comment (petergphillips) [reply Dockerfile]: Agreed, will change on next iteration.  Copied verbatim from a different project I'm afraid.

File: build.gradle (hunk 3 above)
Comment (petergphillips) [reply build.gradle]: Well reading https://github.com/FasterXML/jackson-modules-java8#maven-dependencies it seems to me that the github project is deprecated, but the dependencies are still being pushed to the same place, so think this is okay as is.
